sympy/vector/coordsysrect.py

```diff
@@ -1,19 +1,45 @@
+import types
+from sympy.simplify import simplify
 from sympy.core.basic import Basic
 from sympy.vector.scalar import BaseScalar
-from sympy import eye, trigsimp, ImmutableMatrix as Matrix, Symbol
+from sympy import eye, trigsimp, ImmutableMatrix as Matrix, Symbol, symbols, sqrt, Lambda, Dummy
 from sympy.core.compatibility import string_types, range
 from sympy.core.cache import cacheit
 from sympy.vector.orienters import (Orienter, AxisOrienter, BodyOrienter,
                                     SpaceOrienter, QuaternionOrienter)
-import sympy.vector
 
 
-class CoordSysCartesian(Basic):
+def build_instance_from_lambda(fun=lambda x, y, z: (x, y, z)):
```

Comment:
No, I'm using the ```@SymPyDeprecationWarning```.  I believe that is what @moorepants  suggested.  I still have to figure out how to document it correctly for PR.  

I'll work on the proposal now.  

It should surely be well rounded.  I have checked out the ```sympy.diffgeom``` library, but our coordinate systems will be globally valid, which comes back to the smooth/invertible transformations.  If the user is worried about regions of validity, they will be told to consult the ```sympy.diffgeom``` in our docs.